IMC_WebDev/FromDayToNight-CreatingInteractiveColorPalettesWithJavaScript/blog.md

```diff
@@ -0,0 +1,394 @@
+
+Now, to set up our day and night color palettes, we will set up two functions labelled `day` and `night`:
+
+```js
+...
+<script>
+    ...
+    function day() {
+        
+      }
+    function night() {
+        
+      }
+</script>
+...
+```
+
+For each function, we will need to establish two changes: one for our sun-to-moon switch, and another for our day-to-night color change.
+
+To display our sun during the day and our moon at night, we will retrieve information from our `#sun` and `#moon` elements by using the `setAttribute()` method on our `circle` variable. For the two required parameters, we will set the `id` of this variable to the `sun` and `moon` elements for the day and night color palettes, respectively:
+
+```js
+...
+function day() {
+    circle.setAttribute("id", "sun");
+}
+function night() {
+    circle.setAttribute("id", "moon");
+}
+...
+```
+
+To set our sky colors for the daytime and nighttime, we will accessing the `backgroundColor` attribute of the `body` element through our `sky` variable by appending `.style.backgroundColor` to retrieve and manipulate the CSS properties of this variable. Then, we will set this equal to its respective color of the day and night color palettes in our `colors` array (remember, the index of an array always starts from the number `0`, so `colors[0]` is referring to the daytime color while `colors[1]` is referring to the nighttime color):
+
+```js
+...
+function day() {
+    circle.setAttribute("id", "sun");
+    sky.style.backgroundColor = colors[0];
+}
+function night() {
+    circle.setAttribute("id", "moon");
+    sky.style.backgroundColor = colors[1];
+}
+...
+```
+
+This is what your completed JavaScript code should look like:
+
+```js
+...
+<script>
+    var circle = document.getElementById("sun");
+    var sky = document.getElementById("body");
+    var colors = ["#D5F3FE","#343F8A"];
+
+    function day() {
+        circle.setAttribute("id", "sun");
+        sky.style.backgroundColor = colors[0];
+      }
+      function night() {
+        circle.setAttribute("id", "moon");
+        sky.style.backgroundColor = colors[1];
+      }
+</script>
+...
+```
+
+To tie everything together, we need move back to our HTML code to link our `day()` and `night()` functions to our `day` and `night` buttons, respectively, by adding the `onclick` event to each button:
+
+```html
+...
+<!--HTML-->
+  <body>
+    <button onclick = "day()">day</button>
+    <button onclick = "night()">night</button>
+...
+```
+
+Now, you can interact with your Tiny City. When you click on the `night` button, it should look like this:
+
+![js_code_1](photos/js_code_1.jpg)
+
+
+
+## Conclusion
+
+Congrats on adding an interactive color component to your Tiny City!
+
+Throughout this tutorial, we created and designed buttons in JavaScript, and learned how to stylize and link features to these buttons in HTML and CSS. 
```

Comment:
Please add a little more to the conclusion — even just a sentence. Options include further implementations of what you learned, other features that they can try to add on their own, why these skills are useful in the real world, etc. This helps wrap things up a little more!